File: code/backend/java/core/src/main/java/games/paths/core/port/story/CatalogWriterPort.java
package games.paths.core.port.story;

import java.util.List;

import games.paths.core.model.story.StorySummary;

/**
 * CatalogWriterPort - v0.37.6 outbound port that serialises a story list and
 * stores it where the game website can read it (local dir, S3, …).
 */
public interface CatalogWriterPort {

    /** False when no destination is configured: the export must refuse to run. */
    boolean isConfigured();

    /** Human-readable destination, e.g. a directory path or `s3://bucket`. */
    String target();

    /** Writes the stories as the `GET /api/stories` JSON body; returns the byte count. */
    int write(String relativePath, List<StorySummary> stories);
}

File: code/backend/python/app/core/ports/story/catalog_writer_port.py
from abc import ABC, abstractmethod
from typing import List
from app.core.models.story.story_summary import StorySummary


class CatalogWriterPort(ABC):
    """v0.37.6 — outbound port: stores a story list where the game website reads it."""

    @abstractmethod
    def is_configured(self) -> bool:
        pass

    @abstractmethod
    def target(self) -> str | None:
        pass

    @abstractmethod
    def write(self, relative_path: str, stories: List[StorySummary]) -> int:
        """Writes the `GET /api/stories` JSON body; returns the byte count."""
        pass
